pkg/network/driver_interface.go

```diff
@@ -279,6 +351,42 @@ func resizeDriverBuffer(compareSize int, buffer []uint8) []uint8 {
 	return buffer
 }
 
+// ReadDNSPacket visits a raw DNS packet if one is available.
+func (di *DriverInterface) ReadDNSPacket(visit func([]byte, time.Time) error) (didRead bool, err error) {
+	var bytesRead uint32
+	var key uint32 // returned by GetQueuedCompletionStatus, then ignored
+	var ol *windows.Overlapped
+	err = windows.GetQueuedCompletionStatus(di.dnsIOCP, &bytesRead, &key, &ol, 0)
```

Comment:
Want to note here for future reference.
Using `0` as the timeout makes this a strict polling loop, so we're busy-waiting on DNS packets. 
This is the same mechanism as linux, so revisiting this is probably outside the scope of this PR.
But wanted to call it out just the same